### PR TITLE
[fix]: [CI-3474]: fixed some connector apis

### PR DESCRIPTION
**Pull request by devkimittal**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31668)

### CODE REVIEW

**Comment by devkimittal** [comment]
trigger authorcheck

**Comment by devkimittal** [comment]
trigger ti1

File: 120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java [hunk 1]
```diff
@@ -290,7 +290,6 @@ private void validateTheUpdateRequestIsValid(ConnectorInfoDTO connectorInfo, Str
     validateTheConnectorTypeIsNotChanged(existingConnector.getConnectorType(), connectorInfo.getConnectorType(),
         accountIdentifier, connectorInfo.getOrgIdentifier(), connectorInfo.getProjectIdentifier(),
         connectorInfo.getIdentifier());
-    validateNameIsUnique(connectorInfo, accountIdentifier, existingConnector.getIdentifier());
```
Comment (devkimittal) [inline 120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java:293]: This is inconsistent with other entities i.e. identifier should be unique in a scope but name can be same.

File: 440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java [hunk 2]
```diff
@@ -388,7 +388,7 @@ private ConnectorResponseDTO createInternal(
     if (HARNESS_SECRET_MANAGER_IDENTIFIER.equalsIgnoreCase(connectorRequestDTO.getConnectorInfo().getIdentifier())) {
       log.info("[AccountSetup]:Creating default SecretManager");
     }
-    validateThatAConnectorWithThisNameDoesNotExists(connectorRequestDTO.getConnectorInfo(), accountIdentifier);
```
Comment (devkimittal) [inline 440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java:391]: This is inconsistent with other entities i.e. identifier should be unique in a scope but name can be same.

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/github/GithubUsernameTokenDTO.java [hunk 3]
```diff
@@ -25,6 +26,7 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 @FieldDefaults(level = AccessLevel.PRIVATE)
 @ApiModel("GithubUsernameToken")
+@OneOfField(fields = {"username", "usernameRef"})
```
Comment (devkimittal) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/github/GithubUsernameTokenDTO.java:29]: Exactly one of the fields should be present

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/gitlab/GitlabUsernameTokenDTO.java [hunk 4]
```diff
@@ -25,6 +26,7 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 @FieldDefaults(level = AccessLevel.PRIVATE)
 @ApiModel("GitlabUsernameToken")
+@OneOfField(fields = {"username", "usernameRef"})
```
Comment (devkimittal) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/gitlab/GitlabUsernameTokenDTO.java:29]: Exactly one of the fields should be present

**Comment by devkimittal** [comment]
trigger ti1

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 10 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668)* status: 8 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java, line 293 at r1](https://reviewable.io/reviews/harness/harness-core/31668#-MypYmZIAxoSJZ8hKXMv:-MypYmZJ8W2_AEYOpRWi:bt4cfu1) ([raw file](https://github.com/harness/harness-core/blob/e906abf04b25a69ad5aac42f6719cfab5caa991e/120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java#L293)):*
> ```Java
>         accountIdentifier, connectorInfo.getOrgIdentifier(), connectorInfo.getProjectIdentifier(),
>         connectorInfo.getIdentifier());
>     validateNameIsUnique(connectorInfo, accountIdentifier, existingConnector.getIdentifier());
> ```

@deepakpatankar i remember this being  a req earlier

**Review by aman-harness — APPROVED** [review]
:lgtm:

> Reviewed 10 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668#-:-Mypg1IE5AQvn9cmXIvc:bnfp4nl)* status: :shipit: complete! ✓

**Comment by devkimittal** [comment]
trigger messagecheck

**Review by DeepakPatankar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668)* status: 1 change requests. 1. unresolved discussions. 

___
*[120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java, line 293 at r1](https://reviewable.io/reviews/harness/harness-core/31668#-MypYmZIAxoSJZ8hKXMv:-Myppqfx0-qPGR6jD8Mm:bxsn1zm) ([raw file](https://github.com/harness/harness-core/blob/e906abf04b25a69ad5aac42f6719cfab5caa991e/120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java#L293)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

@deepakpatankar i remember this being  a req earlier
</blockquote></details>

Yes this is a requirement

**Comment by devkimittal** [comment]
___
*[120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java, line 293 at r1](https://reviewable.io/reviews/harness/harness-core/31668#-MypYmZIAxoSJZ8hKXMv:-Mypvr982L3R7vAzNeFk:b-obcr6q) ([raw file](https://github.com/harness/harness-core/blob/e906abf04b25a69ad5aac42f6719cfab5caa991e/120-ng-manager/src/main/java/io/harness/ng/ConnectorServiceImpl.java#L293)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Yes this is a requirement
</blockquote></details>

@deepakpatankar Can you please convey the same to this ticket here? https://harness.atlassian.net/browse/CI-3474
Vikas Naiyar had pointed this out.

**Comment by devkimittal** [comment]
@DeepakPatankar @abhinav-harness,  @vikas-harness has confirmed that names can be same in connectors.

**Review by aman-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668#-:-MyrdbwV9TkkJgulhVAu:bnfp4nl)* status: 1 change requests. 1. unresolved discussions.

**Review by DeepakPatankar — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668#-:-MyrrYay7iTtaFBPkEpC:bnfp4nl)* status: 1 change requests. 1. unresolved discussions.

**Review by devkimittal — COMMENTED** [review]
> Reviewed 10 of 10 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31668#-:-Myu_LQi4fka-iVjfrzu:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**